tree/ntuple/v7/src/RPageStorageDaos.cxx

```diff
@@ -140,30 +142,80 @@ ROOT::Experimental::Detail::RDaosNTupleAnchor::Deserialize(const void *buffer, s
    return result.Unwrap() + 20;
 }
 
-std::uint32_t
-ROOT::Experimental::Detail::RDaosNTupleAnchor::GetSize()
+std::uint32_t ROOT::Experimental::Detail::RDaosNTupleAnchor::GetSize()
 {
-   return RDaosNTupleAnchor().Serialize(nullptr)
-      + ROOT::Experimental::Detail::RDaosObject::ObjClassId::kOCNameMaxLength;
+   return RDaosNTupleAnchor().Serialize(nullptr) +
+          ROOT::Experimental::Detail::RDaosObject::ObjClassId::kOCNameMaxLength;
 }
 
+int FetchNTupleDescriptorBuilder(ROOT::Experimental::Detail::RDaosContainer &cont,
+                                 ROOT::Experimental::Detail::RNTupleDecompressor &decompr,
+                                 RDaosContainerNTupleLocator &loc)
+{
+   std::unique_ptr<unsigned char[]> buffer, zipBuffer;
+   auto &anchor = *loc.fAnchor;
+   int err;
+
+   const auto anchorSize = ROOT::Experimental::Detail::RDaosNTupleAnchor::GetSize();
+   daos_obj_id_t oidMetadata{kOidLowMetadata, static_cast<decltype(daos_obj_id_t::hi)>(loc.GetIndex())};
+
+   buffer = std::make_unique<unsigned char[]>(anchorSize);
+   if ((err = cont.ReadSingleAkey(buffer.get(), anchorSize, oidMetadata, kDistributionKeyDefault, kAttributeKeyAnchor,
+                                  kCidMetadata)))
+      return err;
+
+   anchor.Deserialize(buffer.get(), anchorSize).Unwrap();
+
+   loc.fDescBuilder.SetOnDiskHeaderSize(anchor.fNBytesHeader);
+   buffer = std::make_unique<unsigned char[]>(anchor.fLenHeader);
+   zipBuffer = std::make_unique<unsigned char[]>(anchor.fNBytesHeader);
+   if ((err = cont.ReadSingleAkey(zipBuffer.get(), anchor.fNBytesHeader, oidMetadata, kDistributionKeyDefault,
+                                  kAttributeKeyHeader, kCidMetadata)))
+      return err;
+   decompr.Unzip(zipBuffer.get(), anchor.fNBytesHeader, anchor.fLenHeader, buffer.get());
+   ROOT::Experimental::Internal::RNTupleSerializer::DeserializeHeaderV1(buffer.get(), anchor.fLenHeader,
+                                                                        loc.fDescBuilder);
+
+   loc.fDescBuilder.AddToOnDiskFooterSize(anchor.fNBytesFooter);
+   buffer = std::make_unique<unsigned char[]>(anchor.fLenFooter);
+   zipBuffer = std::make_unique<unsigned char[]>(anchor.fNBytesFooter);
+   if ((err = cont.ReadSingleAkey(zipBuffer.get(), anchor.fNBytesFooter, oidMetadata, kDistributionKeyDefault,
+                                  kAttributeKeyFooter, kCidMetadata)))
+      return err;
+   decompr.Unzip(zipBuffer.get(), anchor.fNBytesFooter, anchor.fLenFooter, buffer.get());
+   ROOT::Experimental::Internal::RNTupleSerializer::DeserializeFooterV1(buffer.get(), anchor.fLenFooter,
+                                                                        loc.fDescBuilder);
+
+   return 0;
+}
 
-////////////////////////////////////////////////////////////////////////////////
+RDaosContainerNTupleLocator LocateNTupleInContainer(ROOT::Experimental::Detail::RDaosContainer &cont,
```

Comment:
Consider making this a (static) member function of the `RDaosContainerNTupleLocator` class.